app/forms/hyrax/forms/resource_form.rb

```diff
@@ -121,18 +121,22 @@ class << self
         #   monograph  = Monograph.new
         #   change_set = Hyrax::Forms::ResourceForm.for(monograph)
         def for(resource)
-          "#{resource.class.name}Form".constantize.new(resource)
-        rescue NameError => _err
-          case resource
-          when Hyrax::AdministrativeSet
-            Hyrax::Forms::AdministrativeSetForm.new(resource)
-          when Hyrax::FileSet
-            Hyrax::Forms::FileSetForm.new(resource)
-          when Hyrax::PcdmCollection
-            Hyrax::Forms::PcdmCollectionForm.new(resource)
+          klass = "#{resource.class.name}Form".safe_constantize
+
+          if klass
```

Comment:
maybe `return  klass.new(resource) if klass`, to shrink the method by a few lines and satisfy rubocop, (i think this is more readable, too)